src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -0,0 +1,325 @@
+using Microsoft.ML.Runtime.Model;
+
+[assembly: LoadableClass(typeof(BinaryPredictionTransformer<IPredictorProducing<float>>), typeof(BinaryPredictionTransformer), null, typeof(SignatureLoadModel),
+    "", BinaryPredictionTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(MulticlassPredictionTransformer<IPredictorProducing<VBuffer<float>>>), typeof(MulticlassPredictionTransformer), null, typeof(SignatureLoadModel),
+    "", MulticlassPredictionTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(RegressionPredictionTransformer<IPredictorProducing<float>>), typeof(RegressionPredictionTransformer), null, typeof(SignatureLoadModel),
+    "", RegressionPredictionTransformer.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.Data
+{
+    public abstract class PredictionTransformerBase<TModel> : IPredictionTransformer<TModel>, ICanSaveModel
+        where TModel : class, IPredictor
+    {
+        protected readonly IHost Host;
+        protected readonly ISchemaBindableMapper BindableMapper;
+        protected readonly ISchema TrainSchema;
+
+        public string FeatureColumn { get; }
+
+        public ColumnType FeatureColumnType { get; }
+
+        public TModel Model { get; }
+
+        public PredictionTransformerBase(IHost host, TModel model, ISchema trainSchema, string featureColumn)
+        {
+            Contracts.CheckValue(host, nameof(host));
+            Host = host;
+            Host.CheckValue(trainSchema, nameof(trainSchema));
+
+            Model = model;
+            FeatureColumn = featureColumn;
+            trainSchema.TryGetColumnIndex(featureColumn, out int col);
+            if (col < 0)
+                throw Host.ExceptSchemaMismatch(nameof(featureColumn), "feature", featureColumn);
+            FeatureColumnType = trainSchema.GetColumnType(col);
+
+            TrainSchema = trainSchema;
+            BindableMapper = ScoreUtils.GetSchemaBindableMapper(Host, model);
+        }
+
+        internal PredictionTransformerBase(IHost host, ModelLoadContext ctx)
+        {
+            Host = host;
+
+            ctx.LoadModel<TModel, SignatureLoadModel>(host, out TModel model, "Model");
+            Model = model;
+
+            // *** Binary format ***
+            // model: prediction model.
+            // stream: empty data view that contains train schema.
+            // id of string: feature column.
+            var ms = new MemoryStream();
+            ctx.TryLoadBinaryStream("TrainSchema", reader =>
+            {
+                reader.BaseStream.CopyTo(ms);
+            });
+
+            ms.Position = 0;
+            var loader = new BinaryLoader(host, new BinaryLoader.Arguments(), ms);
+            TrainSchema = loader.Schema;
+
+            FeatureColumn = ctx.LoadString();
+            TrainSchema.TryGetColumnIndex(FeatureColumn, out int col);
+            if (col < 0)
+                throw Host.ExceptSchemaMismatch(nameof(FeatureColumn), "feature", FeatureColumn);
```

Comment:
>"feature" [](start = 71, length = 9)

DefaultColumnNames.Features maybe across all the code?

Reply:
it's a difference between column name and column role.
Our 'Features' column is our feature column.

---
In reply to: [212679990](https://github.com/dotnet/machinelearning/pull/716#discussion_r212679990) [](ancestors = 212679990)

Reply:
ColumnRole.Feature.Value?

---
In reply to: [212680432](https://github.com/dotnet/machinelearning/pull/716#discussion_r212680432) [](ancestors = 212680432,212679990)